exporter/collector/googlemanagedprometheus/naming.go

```diff
@@ -56,11 +57,23 @@ func GetMetricName(baseName string, metric pmetric.Metric) (string, error) {
 
 // getUnknownMetricSuffix will set the metric suffix for untyped metrics to
 // "/unknown" (eg, for Gauge) or "/unknown:{secondarySuffix}" (eg, "/unknown:counter" for Sum).
+// It also removes the "_total" suffix on an unknown counter, if this suffix was not present in
+// the original metric name before calling prometheus.BuildCompliantName(), which is hacky.
 // It is based on the untyped_prometheus_metric data point attribute, and behind a feature gate.
-func getUnknownMetricSuffix(points pmetric.NumberDataPointSlice, suffix, secondarySuffix string) string {
+func getUnknownMetricName(points pmetric.NumberDataPointSlice, suffix, secondarySuffix, originalName, compliantName string) string {
 	if !untypedDoubleExportFeatureGate.IsEnabled() {
-		return suffix
+		return compliantName + suffix
 	}
+
+	// de-normalize "_total" suffix for counters where not present on original metric name
+	nameTokens := strings.FieldsFunc(
```

Comment:
is strings.HasSuffix not sufficient?  Why this extra strings logic?

Reply:
This is the same code to break it into tokens that the prom library uses, I think to cover cases like: `sample.metric.total`, `foosecondstotal`, `bar_requests_total`